packages/components/input/src/input.ts

```diff
@@ -53,24 +74,54 @@ export class Input extends FormControlMixin(HintMixin(LitElement)) {
   /** Minimum length (number of characters). */
   @property({ type: Number, attribute: 'minlength' }) minLength?: number;
 
+  /** Minimum value. Only applies to number input type.	*/
+  @property({ type: Number, attribute: 'min' }) min?: number;
+
+  /** Maximum value. Only applies to number input type. */
+  @property({ type: Number, attribute: 'max' }) max?: number;
+
+  /** Specifies the interval between legal numbers for an input field. Only applies to number input type */
+  @property({ type: Number, attribute: 'step' }) step?: number;
+
   /** Validation using pattern. */
   @property() pattern?: string;
 
   /** Placeholder text in the input. */
   @property() placeholder?: string;
 
+  /** Whether the input is invalid. */
+  @property({ type: Boolean, reflect: true }) invalid?: boolean;
+
+  /** Whether the input is valid. */
+  @property({ type: Boolean, reflect: true }) valid?: boolean;
+
+  /** Whether the input should get valid styles when is valid. */
+  @property({ type: Boolean, reflect: true }) showValid = false;
+
+  /** Whether the input has focus visible. */
+  @property({ type: Boolean, reflect: true, attribute: 'focus-visible-within' }) focusVisible = false;
+
+  /** Whether you can interact with the input or if it is just a static, readonly display. */
+  @property({ type: Boolean, reflect: true }) readonly?: boolean;
+
+  /** Input size. */
+  @property({ reflect: true }) size: InputSize = 'md';
+
   /**
    * The input type. Only text types are valid here. For other types,
    * see their respective components.
    */
-  @property() type: 'email' | 'number' | 'password' | 'tel' | 'text' | 'url' = 'text';
+  @property() type: 'email' | 'number' | 'tel' | 'text' | 'url' = 'text'; // TODO: password type will be added in the future
 
   /** Custom validators specified by the user. */
   @property({ attribute: false }) validators?: Validator[];
 
   /** The value for the input. */
   @property() value?: string;
 
+  /** @private used for focusVisible */
+  #clicked = false;
```

Comment:
Can you explain this bit?

Reply:
So affairs this goes against the browser heuristics. For example see the demo here: https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible

I really don't like going *against* browser behavior. 😕